core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestUpdateSettingsAction.java

```diff
@@ -63,29 +53,22 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         updateSettingsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", updateSettingsRequest.masterNodeTimeout()));
         updateSettingsRequest.indicesOptions(IndicesOptions.fromRequest(request, updateSettingsRequest.indicesOptions()));
 
-        Settings.Builder updateSettings = Settings.builder();
-        String bodySettingsStr = request.content().utf8ToString();
-        if (Strings.hasText(bodySettingsStr)) {
-            Settings buildSettings = Settings.builder()
-                .loadFromSource(bodySettingsStr, request.getXContentType())
-                .build();
-            for (Map.Entry<String, String> entry : buildSettings.getAsMap().entrySet()) {
-                String key = entry.getKey();
-                String value = entry.getValue();
+        Map<String, Object> settings = new HashMap<>();
+        if (request.hasContent()) {
```

Comment:
this is a nit IMO, but might be cleaner if the `request#contentOrSourceParamParserOrNull` method is used?